### PR TITLE
add CF cluster validation

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/2390

Adds simple structure for checking multiple validation conditions before cluster creation, aborting if any of them fails. Also adds an initial check for existing peering routes in the host accepter VPC route tables.

Still missing `Alerter` changes suggested in https://github.com/giantswarm/giantswarm/issues/2390.

### CODE REVIEW

File: service/resource/cloudformationv2/validate.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+)
+
+type validator func(v1alpha1.AWSConfig) error
+
+func (r *Resource) validateCluster(cluster v1alpha1.AWSConfig) error {
+	validators := []validator{
+		r.validateHostPeeringRoutes,
+	}
```
Comment (rossf7) [inline service/resource/cloudformationv2/validate.go]: Nice, I like that we can extend the validator easily.

File: service/resource/cloudformationv2/validate.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	return nil
+}
+
+func (r *Resource) validateHostPeeringRoutes(cluster v1alpha1.AWSConfig) error {
+	input := &ec2.DescribeRouteTablesInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name: aws.String("route.destination-cidr-block"),
+				Values: []*string{
+					aws.String(cluster.Spec.AWS.VPC.PrivateSubnetCIDR),
```
Comment (rossf7) [inline service/resource/cloudformationv2/validate.go]: Nitpick but we could use a key helper as we use the PrivateSubnetCIDR a few times.

File: service/resource/cloudformationv2/validate.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+				},
+			},
+			&ec2.Filter{
+				Name: aws.String("vpc-id"),
+				Values: []*string{
+					aws.String(cluster.Spec.AWS.VPC.PeerID),
```
Comment (rossf7) [inline service/resource/cloudformationv2/validate.go]: Same here.

File: service/resource/cloudformationv2/validate.go (hunk 2 above)
Comment (fgimenez) [reply service/resource/cloudformationv2/validate.go]: Good point, done, I've updated all the references to the use the new key helpers.

**Pull request merged**